frontends/parsers/p4/p4parser.ypp

```diff
@@ -316,8 +366,127 @@ annotations
 
 annotation
     : "@" name                        { $$ = new IR::Annotation(@1, *$2, {}); }
-    | "@" name "(" expressionList ")" { $$ = new IR::Annotation(@1, *$2, *$4); }
-    | "@" name "(" kvList ")"         { $$ = new IR::Annotation(@1, *$2, *$4); }
+    | "@" name "(" annotationBody ")" { $$ = new IR::Annotation(@1, *$2, *$4); }
+    ;
+
+annotationBody
+    : /* empty */
+        { $$ = new IR::Vector<IR::AnnotationToken>; }
+    | "(" annotationBody ")"
+        { $$ = new IR::Vector<IR::AnnotationToken>;
```

Comment:
This only accepts nested `(`..`)` at the very beginning of the annotationBody.  So something like `@anot( func(7) )` will cause a syntax error.  You need to change this rule to `annotationBody: annotationBody "(" annotationBody ")"`

Reply:
Thanks, good catch. Fixed.